samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -21,6 +21,8 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
```

Comment:
Any particular reason we are using an LRU cache instead of an in mem set here? we are not using anything cache specific ie invalidation

Reply:
We want 2 things:
1. LRU cache of some form - some files tend to stick around for a long time - typically files on higher level in level db. So we want them to be kept around for far longer than files that are merged into the next level or were just created. 
2. Fixed size of the cache - we don't want to balloon the cache size to be too big as we create lots of files over time (most of which are deleted in the next commit). 
I considered using set but I think Guava cache (which we already are using elsewhere in this module) seemed like a a cleaner way to do this.